src/adder.rs

```diff
@@ -1,39 +1,55 @@
+#[path = "./table_header_adder.rs"]
+mod table_header_adder;
+
 use anyhow::{bail, Context, Result};
 use serde_json::{from_str, Value as JValue};
 use toml_edit::{Array, ArrayOfTables, DocumentMut, InlineTable, Item, Table, Value};
 
 use crate::converter::json_to_toml;
 use crate::field_finder::{get_field, DoInsert, TomlValue};
 
-pub fn handle_add(field: &str, value: &str, doc: &mut DocumentMut) -> Result<()> {
-    let mut path_split = field
-        .split('/')
-        .map(|s| s.to_string())
-        .collect::<Vec<String>>();
-
-    let last_field = path_split.pop().context("Path is empty")?;
-
-    let final_field_value =
-        get_field(&path_split, &last_field, DoInsert::Yes, doc).context("Could not find field")?;
-
-    let field_value_json: JValue = from_str(value).context("parsing value field in add request")?;
-
-    let is_inline = matches!(
-        final_field_value,
-        TomlValue::InlineTable(_) | TomlValue::Array(_) | TomlValue::Value(_)
-    );
-
-    let field_value_toml: Item = json_to_toml(&field_value_json, is_inline)
-        .context("converting value in add request from json to toml")?;
-
-    match final_field_value {
-        TomlValue::Table(table) => add_in_table(table, &last_field, field_value_toml),
-        TomlValue::ArrayOfTables(array) => {
-            add_in_array_of_tables(array, &last_field, field_value_toml)
+pub fn handle_add(
+    field: &str,
+    table_header_path: Option<String>,
+    dotted_path: Option<String>,
+    value: &str,
+    doc: &mut DocumentMut) -> Result<()> {
+    match table_header_path {
+        Some(thpath) => {
+            let dpath = dotted_path.context("error: expected dotted_path to add")?;
```

Comment:
can you enforce this in the `Op` enum somehow instead?

Reply:
How?

Reply:
Do you mean like a tagged union? But the struct is used for JSON deserialization, can you use tagged union with JSON/serde? Hmm...

Reply:
https://serde.rs